### ISSUE TITLE
Mypy reports nonsuppressible false positives from OpenCV

### ISSUE THREAD

**Issue by pwuertz** [python/mypy#16152]
[**Bug Report**]

Before OpenCV 4.8, mypy raised errors for not being able to import `cv2`, which I could handle by adding `ignore_missing_imports = true` in the mypy configuration.

Since OpenCV 4.8, I think the [`cv2` package](https://pypi.org/project/opencv-python-headless/) now provides some sort of stubs or changed its import magic. Either way, mypy now apparently considers `cv2` to be importable, but it raises `error: Module has no attribute "X"  [attr-defined]` for every `cv2.X` lookup within an analyzed code base.

Any attempts to silence errors caused by `cv2` by mypy config have failed. Is this because the options I tried only affect errors **raised within** a specified module, whereas I'd require an option for ignoring errors **caused by** a specific module? Does mypy have other options for dealing with such issues?

**To Reproduce**

```python
import cv2
print(cv2.__version__)
# error: Module has no attribute "__version__"  [attr-defined]
```

**Your Environment**

- Mypy version used: 1.5.1
- Mypy configuration options from `pyproject.toml` (and other config files):
```toml
[[tool.mypy.overrides]]
module = "cv2"
follow_imports = "skip"
disable_error_code = "attr-defined"
ignore_errors = true
```
- Python version used: 3.11

**Comment by erictraut** [comment]
The latest version of `cv2` includes type stubs and includes a "py.typed" marker file, which tells a type checker that it should honor the type information provided in the library. Mypy is correctly honoring that type information. For example, the type stubs do not indicate that `__version__` is a public symbol exported from the top-level module, so mypy correctly reports an error if your code attempts to access a symbol by this name.

If you think there are errors in the cv2 type information, you could report the issue to the maintainers of the `opencv-python` package.

In the meantime, here are a few workarounds:
1. Manually delete the "py.typed" marker file from cv2. Mypy will no longer assume that the type information included in the package should be used.
2. Downgrade to cv2 4.7.x.
3. Write your own project-local type stubs for specific cv2 submodules that add the symbols missing from the inlined stubs.
4. Use a `# type: ignore` comment to suppress individual errors.

**Comment by hauntsaninja** [comment]
If you'd like to pretend that cv2 does not have a py.typed, you can also do something like this:
```
[[tool.mypy.overrides]]
module = "cv2.*"
follow_imports = "skip"
follow_imports_for_stubs = true
```
Note that this will make all symbols from cv2 (existing or non-existing) equivalent to Any

**Comment by pwuertz** [comment]
@erictraut Oh, I see! I didn't realize that all the `[attr-defined]` errors I am seeing right now are mostly stemming from a certain subset of symbols currently missing in the opencv stub package (all the CV_X datatype constants for example https://github.com/opencv/opencv/issues/23910). Most of the `cv2` symbols are in fact working perfectly fine with mypy now.

@hauntsaninja Thanks for configuration hint! That's a good workaround for now until the OpenCV type stubs have matured a bit.
